keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class RandomShear(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Randomly shears an image.
+
+    Args:
+        x: float, 2 element tuple, or `None`.  For each augmented image a value is
+            sampled from the provided range.  If a float is passed, the range is
+            interpreted as `(0, x)`.  If `None` is passed, no shear occurs on the X
+            axis.  Defaults to `None`.
+        y: float, 2 element tuple, or `None`.  For each augmented image a value is
+            sampled from the provided range.  If a float is passed, the range is
+            interpreted as `(0, y)`.  If `None` is passed, no shear occurs on the Y
+            axis.  Defaults to `None`.
+        interpolation: interpolation method used in the `ImageProjectiveTransformV3` op.
+             Supported values can be found in the `ImageProjectiveTransformV3`
+             documentation.  Defaults to `"bilinear"`.
+        fill_mode: fill_mode in the `ImageProjectiveTransformV3` op.
+             Supported values can be found in the `ImageProjectiveTransformV3`
+             documentation.  Defaults to `"reflect"`.
+        fill_value: fill_value in the `ImageProjectiveTransformV3` op.
+             Supported values can be found in the `ImageProjectiveTransformV3`
+             documentation.  Defaults to `0.0`.
+    """
+
+    def __init__(
+        self,
+        x=None,
+        y=None,
+        interpolation="bilinear",
+        fill_mode="reflect",
+        fill_value=0.0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if isinstance(x, float):
+            x = (0, x)
+        if isinstance(y, float):
+            y = (0, y)
+        if x is None and y is None:
+            warnings.warn(
```

Comment:
Should this be an error instead?

Reply:
No - because some pipelines may gridsearch over hyperparameters and not want to special case remove the layer from their codebase.  Or perhaps users will pass `x_shear` as a flag, and the code base will not know if it is `None` or a value.

I say we keep it as a warning in this case - the COCO metrics had a similar case with sample weights.